code/processes/dqe.q

```diff
@@ -1,34 +1,34 @@
-tickerplanttypes:@[value;`tickerplanttypes;`tickerplant];     /-list of tickerplant types to try and make a connection to
-replaylog:@[value;`replaylog;0b];                             /-replay the tickerplant log file
-tpconnsleepintv:@[value;`tpconnsleepintv;10];                 /-default wait time before trying to reconnect to discovery/TP
-/- end of default parameters
-
-subscribe:{[]
-  if[count s:.sub.getsubscriptionhandles[tickerplanttypes;();()!()];
-  .lg.o[`subscribe;"found available tickerplant, attempting to subscribe"];
-  /-set the date that was returned by the subscription code i.e. the date for the tickerplant log file
-  /-and a list of the tables that the process is now subscribing for
-  subinfo: .sub.subscribe[subscribeto;subscribesyms;schema;replaylog;first s];
-  /-setting subtables and tplogdate globals
-  .dqe,:subinfo]
-  }
-
 init:{
-  .servers.CONNECTIONS:distinct .servers.CONNECTIONS, .dqe.tickerplanttypes;
   .lg.o[`init;"searching for servers"];
-  .servers.startup[];
-  .servers.startupdependent[tickerplanttypes;tpconnsleepintv];
-  subscribe[];
+  .servers.startup[];                                                                                           /- Open connection to discovery
   }
 
-tableExists:{x in tables[]};                                  /-function to check for table, param is table name as a symbol
+gethandles:{exec procname,proctype,w from .servers.SERVERS where (procname in x) | (proctype in x)};
+
+tableexists:{x in tables[]};                                                                                    /- function to check for table, param is table name as a symbol
+
+runcheck:{[fn;vars;rs]                                                                                          /- function used to send other function to test processes
+  fncheck:` vs fn;
+  if [not fncheck[2] in key value .Q.dd[`;fncheck 1];                                                           /- run check to make sure passed in function exists
+    .lg.e[`function;"Function ",(string fn)," doesn't exist"];
+    :()];
+
+  rs:(),rs;                                                                                                     /- set rs to a list
+  h:.dqe.gethandles[rs];                                                                                        /- check if processes exist and are valid
+  
+  missingproc:rs where not rs in raze h`procname`proctype;                                                      /- check all process exist
+  if[0<count missingproc;.lg.e[`process;(", "sv string missingproc)," processes are not connectable"]];
+
+  if[0=count h;.lg.e[`handle;"cannot open handle to any given processes"];:()];                                 /- check if any handles exist, if not exit function
+  ans:{[funct;vars;hand].async.postback[hand;(funct;vars);`.dqe.showresult]}[value fn;vars]'[h[`w]]             /- send function with variables down handle
```

Comment:
This may not actually return, can fix in next PR.

Reply:
I had tested it before and it seemed fine.
```
q).dqe.runcheck[`.dqe.tableexists;`quote;`rdb`hdb1`bbo`hfd]
2019.11.22D14:45:07.558743000|homer|dqe|dqe1|ERR|process|bbo, hfd processes are not connectable
,1b
,1b
q)1b
1b
```